### PR TITLE
Minor: name some constant values in arrow writer, parquet writer

### PR DESCRIPTION
**Pull request by alamb**
## Which issue does this PR close?

Follow on to https://github.com/apache/arrow-datafusion/pull/8608

## Rationale for this change

The presence of numbers in the source code was not immediately obvious to me and I think it is clearer when there are comments to guide understanding. See  https://github.com/apache/arrow-datafusion/pull/8608#discussion_r1435356536

## What changes are included in this PR?

Give two constants a name and add some docstrings to explain what they do

### CODE REVIEW

**Comment by devinjdangelo** [comment]
LGTM. I believe the parallel writer has the same two constants as well.

File: datafusion/core/src/datasource/file_format/arrow.rs [hunk 1]
```diff
@@ -193,6 +193,13 @@ impl DisplayAs for ArrowFileSink {
     }
 }
 
+/// Initial writing buffer size. Note this is just a size hint for efficiency. It
+/// will grow beyond the set value if needed.
+const INITIAL_BUFFER_BYTES: usize = 1048576;
```
Comment (comphead) [inline datafusion/core/src/datasource/file_format/arrow.rs:198]: Thanks for documenting that, yesterday was pondering what those values exactly mean


there are more to go

https://github.com/apache/arrow-datafusion/blob/main/datafusion/core/src/datasource/file_format/parquet.rs#L1028

https://github.com/apache/arrow-datafusion/blob/main/datafusion/core/src/datasource/file_format/parquet.rs#L1007


https://github.com/apache/arrow-datafusion/blob/main/datafusion/core/src/datasource/file_format/parquet.rs#L683

Comment (alamb) [reply datafusion/core/src/datasource/file_format/arrow.rs:198]: Filed follow on PR https://github.com/apache/arrow-datafusion/pull/8656

File: datafusion/core/src/datasource/file_format/parquet.rs [hunk 2]
```diff
@@ -75,6 +75,17 @@ use crate::physical_plan::{
     Statistics,
 };
 
+/// Size of the buffer for [`AsyncArrowWriter`].
+const PARQUET_WRITER_BUFFER_SIZE: usize = 10485760;
```
Comment (alamb) [inline datafusion/core/src/datasource/file_format/parquet.rs:79]: I don't know why these values are (slightly) different, but I figured we could start by keeping them the same and then could unify them as a follow on if needed

**Review by andygrove — APPROVED** [review]

**Pull request merged**